src/components/ExploreMovieList.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { SyntheticEvent, useState, useEffect } from "react";
+
+import {
+  Movie,
+  MoviePreference,
+  PreferenceType,
+  SetMovies,
+  User
+} from "../types";
+
+import { Card } from "semantic-ui-react";
+
+import ExploreMovieCard from "./ExploreMovieCard";
+
+import {
+  addMoviePreference,
+  fetchMovies,
+  updateMoviePreferenceAsync
+} from "../network/requests";
+
+const ExploreMovieList = (props: {
+  user: User;
+  movies: Array<Movie>;
+  fetchExploreMovies: () => void;
+  setMovies: SetMovies;
+}): JSX.Element => {
+  const [userMovies, setUserMovies] = useState<Array<MoviePreference>>([]);
+  const doFetchUserMovies = () => fetchMovies(props.user, setUserMovies);
+  useEffect(props.fetchExploreMovies, []);
+  useEffect(doFetchUserMovies, []);
+
+  const userMoviePreferenceMap = userMovies.reduce(
+    (acc: any, moviePreference: MoviePreference) => {
```

Comment:
oooh the typing here could be fun - I think `acc` can be `{[string]: PreferenceType}`